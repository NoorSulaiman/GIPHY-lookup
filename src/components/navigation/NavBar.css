
#topNav{
    background-color: #683caf;
    height: 100px;
}

#topNav{
    color: #fff;
    position: relative;
    perspective: 1000px;
    background-image: linear-gradient(28deg, #472086, #4a228a);
    overflow: hidden;
}

#topNav::before{
    content: '';
    position: absolute;
    display: block;
    width: 100%;
    height: 100%;
    left: 58%;
    top: 0;
    transform: skew(-20deg);
    background-image: linear-gradient(0deg, #5c31a3, #4a218b);
    box-shadow: 0 0 60px 5px rgba(0, 0, 0, 0.1);
}

#navLinksContainer{
    margin-left: 20%;
    margin-top: auto;
    margin-bottom: auto;
    display: flex;
    height: 50PX;   
    justify-content: space-between;
}

#giphylookup{
    font-family: 'Comfortaa', cursive;
    font-size: 1.7em;
    margin: auto 0px auto 0%;
}

.ui.menu a.menuItem{
    color: rgb(197, 194, 194);
    font-size: 1.2em;
    position: relative;
    display: block;
    border-bottom: none;
    transition: all .2s;
    text-decoration: none;
}
.ui.menu a.menuItem:before {
    content: none;
    
}
.ui.menu a.menuItem:hover{
background-color:transparent;
color: #fff;
}

.ui.menu a.menuItem::after{
    content: "";
    display: block;
    bottom: 0;
    height: 1px;
    background: #fff;
    width: 0%;
    left: 50%;
    transform: translateX(-50%);
    transition: all 0.2s;
    position: absolute;
}

.ui.menu a.menuItem:hover::after{
    width:100%;
}