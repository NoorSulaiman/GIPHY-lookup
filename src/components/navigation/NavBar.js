import React from 'react';
import { Menu } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import './NavBar.css';

const NavBar = () => (
    <Menu size="large" id="topNav">
        <div id='navLinksContainer'>
            <Menu.Header id="giphylookup">
                GIPHY Lookup
        </Menu.Header>
            <Menu.Item className="menuItem" as={Link} to="/">
                Home
        </Menu.Item>
            <Menu.Item className="menuItem" as={Link} to="/favorites">
                Favorites
        </Menu.Item>
        </div>
    </Menu>
);

export default NavBar;