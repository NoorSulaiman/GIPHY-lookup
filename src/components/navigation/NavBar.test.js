import React from 'react';
import { shallow } from 'enzyme';
import NavBar from './NavBar';

it('should render NavBar correctly with given strings', () => {
    const component = shallow(<NavBar />);
    expect(component).toMatchSnapshot();
});