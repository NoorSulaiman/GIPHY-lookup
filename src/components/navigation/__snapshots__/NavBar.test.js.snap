// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NavBar correctly with given strings 1`] = `
<Menu
  id="topNav"
  size="large"
>
  <div
    id="navLinksContainer"
  >
    <MenuHeader
      id="giphylookup"
    >
      GIPHY Lookup
    </MenuHeader>
    <MenuItem
      as={[Function]}
      className="menuItem"
      to="/"
    >
      Home
    </MenuItem>
    <MenuItem
      as={[Function]}
      className="menuItem"
      to="/favorites"
    >
      Favorites
    </MenuItem>
  </div>
</Menu>
`;
